keyboards/dozen0/keymaps/default/keymap.c

```diff
@@ -15,20 +15,110 @@
  */
 #include QMK_KEYBOARD_H
 
+#ifdef RGBLIGHT_ENABLE
+//Following line allows macro to read current RGB settings
+extern rgblight_config_t rgblight_config;
+#endif
+
+enum layer_number {
+    _BASE = 0,
+    _LOWER,
+    _RAISE,
+    _ADJUST
+};
+
+enum custom_keycodes {
+  BASE = SAFE_RANGE,
+  LOWER,
+  RAISE,
+  ADJUST,
+  RGBRST
+};
 
 const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
-  [0] = LAYOUT(
-    LCTL(KC_X), LCTL(KC_C), LCTL(KC_V), KC_UP, KC_DEL, KC_BSPC, \
-    KC_LCTRL, KC_LSFT, KC_LEFT, KC_DOWN, KC_RIGHT, KC_ENT \
+  /* Base
+   * ,-----------------------------------------.
+   * |  Cut | Copy |Paste |  Up  |Delete| Bksp |
+   * |      |      |      |      |      |Raise |
+   * |------+------+------+------+------+------|
+   * | Ctrl |Shift | Left | Down |Right |Enter |
+   * |      |      |      |      |      |Lower |
+   * `-----------------------------------------'
+   */
+	[_BASE] = LAYOUT(
+    LCTL(KC_X), LCTL(KC_C), LCTL(KC_V), KC_UP, KC_DEL, LT(_RAISE,KC_BSPC), \
+    KC_LCTL, KC_LSFT, KC_LEFT, KC_DOWN, KC_RGHT, LT(_LOWER,KC_ENT) \
+  ),
+
+  /* Lower
+   * ,-----------------------------------------.
+   * |      |      |      | Page |      |      |
+   * |      |      |      |  Up  |      |      |
+   * |------+------+------+------+------+------|
+   * |      |      | Home | Page | End  |      |
+   * |      |      |      | Down |      |      |
+   * `-----------------------------------------'
+   */
+  [_LOWER] = LAYOUT(
+    _______, _______, _______, KC_PGUP, _______, _______, \
+    _______, _______, KC_HOME, KC_PGDN, KC_END, _______ \
+  ),
+
+  /* Raise
+   * ,-----------------------------------------.
+   * |      |      |      |      |      |      |
+   * |      |      |      |      |      |      |
+   * |------+------+------+------+------+------|
+   * |      |      |      |      |      |      |
+   * |      |      |      |      |      |      |
+   * `-----------------------------------------'
+   */
+  [_RAISE] = LAYOUT(
+    _______, _______, _______, _______, _______, _______, \
+    _______, _______, _______, _______, _______, _______ \
   ),
+
+  /* Adjust
+   * ,-----------------------------------------.
+   * | RGB  | RGB  | RGB  | RGB  | RGB  |      |
+   * |Toggle|Mode+ | Hue+ | Sat+ | Val+ |      |
+   * |------+------+------+------+------+------|
+   * | RGB  | RGB  | RGB  | RGB  | RGB  |      |
+   * |Reset |Mode- | Hue- | Sat- | Val- |      |
+   * `-----------------------------------------'
+   */
+  [_ADJUST] = LAYOUT(
+    RGB_TOG, RGB_MOD, RGB_HUI, RGB_SAI, RGB_VAI, _______, \
+    RGBRST, RGB_RMOD, RGB_HUD, RGB_SAD, RGB_VAD, _______ \
+  )
 };
 
-bool process_record_user(uint16_t keycode, keyrecord_t *record) {
-  return true;
+uint32_t layer_state_set_user(uint32_t state) {
```

Comment:
`layer_state_set_user` is typedef'd now. (Reduces firmware size if there are only a few layers.)

```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
```